lib/pages/survey/survey_controller.dart

```diff
@@ -1,15 +1,25 @@
 import 'package:get/get.dart';
+import 'package:optional/optional.dart';
+import 'package:survey/models/survey.dart';
 import 'package:survey/navigator/const.dart';
+import 'package:survey/use_cases/base_use_case.dart';
 import 'package:survey/use_cases/get_survey_detail_use_case.dart';
 
 class SurveyController extends GetxController {
+  final Rx<Optional<Survey>> _survey = Optional<Survey>.empty().obs;
+
+  Optional<Survey> get optionalSurvey => _survey.value;
+
   @override
   void onInit() {
     super.onInit();
 
     final surveyId = Get.arguments[DATA_SURVEY_ID];
     // TODO: remove this when integrate
     final getSurveyUseCase = Get.find<GetSurveyDetailUseCase>();
-    getSurveyUseCase.call(surveyId).then((value) => print("Value: $value"));
+    getSurveyUseCase.call(surveyId).then((result) => {
+          if (result is Success<Survey>)
+            {_survey.value = Optional.of(result.value)}
```

Comment:
Could you please re-format here a bit? 😂

Reply:
hmmm the auto-format turned it into this...